src/util/route.js

```diff
@@ -6,13 +6,13 @@ export function isSameRoute (a: Route, b: ?Route): boolean {
   } else if (a.path && b.path) {
     return (
       a.path === b.path &&
-      a.hash === b.hash &&
+      (a.hash || '') === (b.hash || '') &&
```

Comment:
By definition, isSameRoute's parameter `a` and `b` is type `Route`, whose `hash` is a required string field, but sometimes b.hash is undefined. So there may be some bug flow wasn't able to discover.